replication.go

```diff
@@ -51,7 +51,11 @@ import (
 
 var defaultReplication = 6
 
-const defaultContentSizeLimit = 34_000_000_000
+// maxFilecoinContentSizeLimit = 31GiB512MiB
+const maxFilecoinContentSizeLimit = 33_822_867_456
```

Comment:
>  // maxFilecoinContentSizeLimit = 31GiB512MiB 

@en0ma this is incorrect. The maximum byte-sized-payload you can get into a 32GiB filecoin sector is: `32GiB / 128 * 127` which comes out to `34,091,302,912` bytes. The value of `34,000,000,000` was chosen to roughly **over**-account for IPLD overhead within the serialized dag. Dropping it further is not warranted.

As a "living example" - this car file weighing at `34,091,006,682` bytes is replicated multiple times within filecoin: https://cargo.web3.storage/deal-cars/bafybeifaow2p3qrzndvr4dzvms7fun3e3bijuwkwrmzou4zupatyennufy_baga6ea4seaqi3u2amdhq6dba6kdaccbevfcbqskvqkutdfphfjsa7ntitll7ajy.car

Reply:
@ribasushi I have LOTS some smaller `31.86GiB` cars that aren't uploading on estuary, or even some `31.73GiB` that are not working.

I belive the biggest issue is that the serialisation overhead depends with the number of blocks.

So a .car that stores a few big blocks would allow a limit really close to `32GiB` while one of lots of small block need far more space for the serialisation overhead.

Reply:
I think the correct solution would be trying to do at least the first step of the CARCompPiece step, to get the final size of the piece, and then decide depending of how big it is (if the piece is above 32GiB or not).

Reply:
@Jorropo I am strictly pointing out how the filecoin chain works. The 31.75 is the maximum value of the thing you run commp over. This is all after the IPLD encoding and spitting out as car. 31.75 - 34000000000 gives you 91MiB of overhead for IPLD ( which is way too much already ).

Reply:
Thanks for the details @Jorropo  @ribasushi. I believe I have a better understanding now. I agree that we should do the check of `34000000000` only after the `CARCompPiece` step, and not before.

Reply:
@en0ma sorry, need to clarify here, details are important.

> we should do the check of 34000000000 only after the CARCompPiece step

No. The check of what you feed into `CARCompPiece` is for `34,091,302,912` bytes, as what you run `commp` over is literally what you put into the sector payload - there is no other variable wrapping overhead.